lib/tpm2_session.c

```diff
@@ -156,6 +160,10 @@ tpm2_session *tpm2_session_new(TSS2_SYS_CONTEXT *sapi_context,
 
     session->internal.nonceNewer.size = session->input->nonce_caller.size;
 
+    if (!sapi_context) {
```

Comment:
Isn't this a bug in the caller? Why would a tool want to start a session without a SAPI context? 

I think this check should be at the beginning of the function and just return NULL. Or maybe I'm misunderstanding something?

Reply:
I think this was just some dumbness on my part.

Reply:
ahh tpm2_session_restore() creates a session context but does not need to start the authsession. This essentially lets you create a session context without calling startauthsession().